provision/README.md

```diff
@@ -1,43 +1,142 @@
+| MF_PROVISION_SERVER_KEY             | Mainflux gRPC secure server key                   | ""                               |
+| MF_PROVISION_MQTT_URL               | Mainflux MQTT adapter URL                         | "http://localhost:1883"          |
+| MF_PROVISION_USERS_LOCATION         | Users service URL                                 | "http://locahost"                |
+| MF_PROVISION_THINGS_LOCATION        | Things service URL                                | "http://localhost"               |
+| MF_PROVISION_LOG_LEVEL              | Service log level                                 | "http://localhost"               |
+| MF_PROVISION_HTTP_PORT              | Service listening port                            | "8091"                           |
+| MF_PROVISION_USER                   | Mainflux user username                            | "test@example.com"               |
+| MF_PROVISION_PASS                   | Mainflux user password                            | "password"                       |
+| MF_PROVISION_BS_SVC_URL             | Mainflux Bootstrap service URL                    | http://localhost/things/configs" |
+| MF_PROVISION_BS_SVC_WHITELIST_URL   | Mainflux Bootstrap service whitelist URL          | "http://localhost/things/state"  |
+| MF_PROVISION_CERTS_SVC_URL          | Certificats service URL                           | "http://localhost/certs"         |
+| MF_PROVISION_X509_PROVISIONING      | Should X509 client cert be provisioned            | "false"                          |
+| MF_PROVISION_BS_CONFIG_PROVISIONING | Should thing config be saved in Bootstrap service | "true"                           |
+| MF_PROVISION_BS_AUTO_WHITEIST       | Should thing be auto whitelisted                  | "true"                           |
+| MF_PROVISION_BS_CONTENT             | Bootstrap service content                         | "{}"                             |
+
+By default, call to `/mapping` endpoint will create one thing and two channels (`control` and `data`) and connect it. If there is a requirement for different provision layout we can use [config](docker/configs/config.toml) file in addition to environment variables. For the purposes of running provision as an add-on in docker composition environment variables seems more suitable. Environoment variables are set in [.env](.env).  
+Configuration can be specified in [config.toml](configs/config.toml). Config file can specify all the settings that environment variables can configure and in addition
+`/mapping` endpoint provision layout can be configured.
+
+In `config.toml` we can enlist array of things and channels that we want to create and make connections between them which we call provision layout.
+Metadata can be whatever suits your needs except that at least one thing needs to have `externalID` (which is populated with value from [request](#example)).
```

Comment:
`external_id`

Reply:
this how it is in thing metadata

Reply:
Then it have to be also fixed in thing metadata

Reply:
this requires additional issue since bootstrap service needs to be changed as well

Reply:
Why? https://github.com/mainflux/mainflux/blob/master/bootstrap/api/responses.go#L73